specs/protocol/isthmus/exec-engine.md

```diff
@@ -63,6 +80,36 @@ places a burden on users of the system in a post-fault-proofs world, where:
 Placing the [`L2ToL1MessagePasser`][l2-to-l1-mp] account storage root in the `withdrawalsRoot` field alleviates this burden
 for users and protocol participants alike, allowing them to propose and verify other proposals with lower operating costs.
 
+#### Genesis Block
+
+If isthmus is active at the genesis block, the withdrawals root is the empty withdrawals root, regardless of L2 state.
+
+#### State Processing
+
+At the time of state processing, the header for which transactions are being validated should not make it's `withdrawalsRoot`
+available to the EVM/application layer.
+
+#### P2P
+
+During sync, the block body withdrawal hash is computed from the withdrawals list in the block body.
+When verifying the final block that is synced, if Isthmus is active at such a block's header,
+we expect that the computed withdrawal hash must match the hash of an empty list of withdrawals.
+The header `withdrawalsRoot` MPT hash can be any non-null value in this case. Verification of the
+final block is performed after a header is available so that the header's timestamp can be checked
+to see if Isthmus is active.
+
+#### Backwards Compatibility Considerations
+
+Beginning at Canyon (which includes Shanghai hardfork support) and prior to Isthmus activation,
+the `withdrawalsRoot` field is set to the MPT root of an empty withdrawals list. This is the
+same root as an empty storage root. The withdrawals are captured in the L2 state, however
+they are not reflected in the `withdrawalsRoot`. Hence, prior to Isthmus activation,
+even if a `withdrawalsRoot` is present and a MPT root is present in the header, it should not be used.
+Any implementation that calculates output root should be careful not to use the header `withdrawalsRoot`.
+
+After Isthmus activation, if there was never any withdrawal contract storage, a MPT root of an empty list
+can be set as the `withdrawalsRoot`
```

Comment:
unless the block is genesis (though I think we should fix that).

Reply:
no longer relevant as genesis header withdrawalsRoot shall also be an MPT root of an empty list (we decided not to special case the genesis withdrawalsRoot).